[qpid-proton] build failure

**Host Environment**
- OS: Linux
- Compiler: `clang version 10.0.0-4ubuntu1`

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install qpid-proton`

on a machine where the [gold linker](https://en.wikipedia.org/wiki/Gold_(linker)) is not installed.

**Failure logs**
```shell
-- Building x64-linux-dynamic-dbg
CMake Error at /home/jenkins/product/debug/vcpkg/scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: /home/klalumiere/product/debug/vcpkg/downloads/tools/cmake-3.22.2-linux/cmake-3.22.2-linux-x86_64/bin/cmake --build . --config Debug --target install -- -v -j17
    Working Directory: /home/jenkins/product/debug/vcpkg/buildtrees/qpid-proton/x64-linux-dynamic-dbg
    See logs for more information:
      /home/jenkins/product/debug/vcpkg/buildtrees/qpid-proton/install-x64-linux-dynamic-dbg-out.log

Call Stack (most recent call first):
  /home/jenkins/product/debug/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:111 (vcpkg_execute_build_process)
  /home/jenkins/product/debug/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  /home/jenkins/product/debug/vcpkg/buildtrees/versioning_/versions/qpid-proton/109a9bb4fac5d5c5a52cd620d7ddb22da6fe47f7/portfile.cmake:23 (vcpkg_cmake_install)
  /home/jenkins/product/debug/vcpkg/scripts/ports.cmake:145 (include)


Error: Building package qpid-proton:x64-linux-dynamic failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+qpid-proton
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[qpid-proton]+Build+error
including:
  package: qpid-proton[core]:x64-linux-dynamic -> 0.32.0#3
    vcpkg-tool version: 2022-03-09-1affd32f93b299d5a907816c328ca3ededb73a7e
    vcpkg-scripts version: 18433251d 2022-03-24 (3 weeks ago)

Additionally, attach any relevant sections from the log files above.
-- Running vcpkg install - failed
CMake Error at debug/vcpkg/scripts/buildsystems/vcpkg.cmake:841 (message):
  vcpkg install failed.  See logs for more information:
  /home/jenkins/product/debug/vcpkg-manifest-install.log
Call Stack (most recent call first):
  /usr/share/cmake-3.21/Modules/CMakeDetermineSystem.cmake:124 (include)
  CMakeLists.txt:24 (project)


CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
```

and

```shell
[76/268] : && /usr/bin/clang -fPIC -fPIC -g  -flto=thin -shared -Wl,-soname,libqpid-proton-core.so.10 -o c/libqpid-proton-core.so.10.10.0 c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/object.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/list.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/map.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/string.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/iterator.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/record.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/init.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/memory.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/logger.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/util.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/error.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/buffer.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/types.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/framing.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/codec.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/decoder.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/encoder.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/dispatcher.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/connection_driver.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/engine.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/event.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/autodetect.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/transport.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/message.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/sasl/sasl.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/sasl/default_sasl.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/sasl/cyrus_stub.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/ssl/openssl.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/compiler/gcc/start.c.o  -Wl,-rpath,/home/jenkins/product/debug/vcpkg_installed/x64-linux-dynamic/debug/lib:  /home/jenkins/product/debug/vcpkg_installed/x64-linux-dynamic/debug/lib/libssl.so  /home/jenkins/product/debug/vcpkg_installed/x64-linux-dynamic/debug/lib/libcrypto.so  -lpthread && :
FAILED: c/libqpid-proton-core.so.10.10.0 
: && /usr/bin/clang -fPIC -fPIC -g  -flto=thin -shared -Wl,-soname,libqpid-proton-core.so.10 -o c/libqpid-proton-core.so.10.10.0 c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/object.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/list.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/map.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/string.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/iterator.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/object/record.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/init.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/memory.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/logger.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/util.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/error.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/buffer.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/types.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/framing.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/codec.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/decoder.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/encoder.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/dispatcher.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/connection_driver.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/engine.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/event.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/autodetect.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/transport.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/core/message.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/sasl/sasl.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/sasl/default_sasl.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/sasl/cyrus_stub.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/ssl/openssl.c.o c/CMakeFiles/qpid-proton-core-objects.dir/src/compiler/gcc/start.c.o  -Wl,-rpath,/home/jenkins/product/debug/vcpkg_installed/x64-linux-dynamic/debug/lib:  /home/jenkins/product/debug/vcpkg_installed/x64-linux-dynamic/debug/lib/libssl.so  /home/jenkins/product/debug/vcpkg_installed/x64-linux-dynamic/debug/lib/libcrypto.so  -lpthread && :
/usr/bin/ld: /usr/lib/llvm-10/bin/../lib/LLVMgold.so: error loading plugin: /usr/lib/llvm-10/bin/../lib/LLVMgold.so: cannot open shared object file: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Additional context**
The problem is due to the flag `-flto=thin`, a flag added by default by qpid-proton build system that, according to [clang documentation](https://clang.llvm.org/docs/ThinLTO.html), can require gold. I'll create a pull request that fixes the issue.

--- a/ports/qpid-proton/portfile.cmake
+++ b/ports/qpid-proton/portfile.cmake
@@ -15,11 +15,12 @@ vcpkg_from_github(
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        -DPYTHON_EXECUTABLE=${PYTHON3}
-        -DLIB_SUFFIX=
         -DBUILD_GO=no
-        -DENABLE_JSONCPP=ON
         -DCMAKE_DISABLE_FIND_PACKAGE_CyrusSASL=ON
+        -DENABLE_JSONCPP=ON
+        -DENABLE_LINKTIME_OPTIMIZATION=OFF
+        -DLIB_SUFFIX=
+        -DPYTHON_EXECUTABLE=${PYTHON3}
 )
 
 vcpkg_cmake_install()

--- a/ports/qpid-proton/vcpkg.json
+++ b/ports/qpid-proton/vcpkg.json
@@ -3,6 +3,7 @@
   "version": "0.32.0",
   "port-version": 5,
   "description": "Qpid Proton is a high-performance, lightweight messaging library.",
+  "license": "Apache-2.0",
   "homepage": "https://github.com/apache/qpid-proton",
   "dependencies": [
     "jsoncpp",

--- a/ports/qpid-proton/vcpkg.json
+++ b/ports/qpid-proton/vcpkg.json
@@ -3,8 +3,8 @@
   "version": "0.32.0",
   "port-version": 5,
   "description": "Qpid Proton is a high-performance, lightweight messaging library.",
-  "license": "Apache-2.0",
   "homepage": "https://github.com/apache/qpid-proton",
+  "license": "Apache-2.0",
   "dependencies": [
     "jsoncpp",
     {

--- a/ports/qpid-proton/vcpkg.json
+++ b/ports/qpid-proton/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "qpid-proton",
   "version": "0.32.0",
-  "port-version": 4,
+  "port-version": 5,
   "description": "Qpid Proton is a high-performance, lightweight messaging library.",
   "homepage": "https://github.com/apache/qpid-proton",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5646,7 +5646,7 @@
     },
     "qpid-proton": {
       "baseline": "0.32.0",
-      "port-version": 4
+      "port-version": 5
     },
     "qscintilla": {
       "baseline": "2.12.0",

--- a/versions/q-/qpid-proton.json
+++ b/versions/q-/qpid-proton.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "2d43828eb37839563c86ce72da4c61a21671a5fe",
+      "git-tree": "bc437ca04548b4ca544af845a2e2327010aa3281",
       "version": "0.32.0",
       "port-version": 5
     },

--- a/versions/q-/qpid-proton.json
+++ b/versions/q-/qpid-proton.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "1b9b8fa3e5b7b6759cb19a7ed211c3720738dc30",
+      "git-tree": "2d43828eb37839563c86ce72da4c61a21671a5fe",
       "version": "0.32.0",
       "port-version": 5
     },

--- a/versions/q-/qpid-proton.json
+++ b/versions/q-/qpid-proton.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "1b9b8fa3e5b7b6759cb19a7ed211c3720738dc30",
+      "version": "0.32.0",
+      "port-version": 5
+    },
+    {
       "git-tree": "be327f08d64834a36d80a140832abbbc66e67207",
       "version": "0.32.0",
       "port-version": 4

